client/me/security-ssh-key/update-ssh-key-form.tsx

```diff
@@ -0,0 +1,126 @@
+import { Button, FormInputValidation } from '@automattic/components';
+import styled from '@emotion/styled';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import i18n from 'i18n-calypso';
+import { ChangeEvent, useReducer, useState } from 'react';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import * as SSHKeyCard from 'calypso/components/ssh-key-card';
+import TextareaAutosize from 'calypso/components/textarea-autosize';
+import { SSH_KEY_FORMATS } from './use-ssh-key-query';
+
+const PUBLIC_SSH_KEY_INPUT_ID = 'public_ssh_key';
+
+const initialState = {
+	touched: false,
+	isValid: false,
+	value: '',
+};
+
+const sshKeyValidation = new RegExp( `^(?:${ SSH_KEY_FORMATS.join( '|' ) })\\s.+` );
+
+type ReducerAction = { type: 'setValue'; value: string };
+
+const sshKeyFormReducer = ( state = initialState, action: ReducerAction ): typeof initialState => {
+	switch ( action.type ) {
+		case 'setValue':
+			return {
+				touched: true,
+				isValid: sshKeyValidation.test( action.value ),
+				value: action.value,
+			};
+
+		default:
+			return state;
+	}
+};
+
+const NewSSHFormFieldContainer = styled.div`
+	margin-top: 20px;
+`;
+
+interface UpdateSSHKeyFormProps {
+	updateSSHKey( args: { name: string; key: string } ): void;
+	isUpdating: boolean;
+	updateText?: string;
+	oldSSHFingerprint: string;
+	userLogin: string;
+	keyName: string;
+}
+
+export const UpdateSSHKeyForm = ( {
```

Comment:
It seems like there's a lot of duplication between this and `<AddSSHKeyForm>`.

Is there any way we could consolidate both to a more generic `<SSHKeyForm>`?